src/Toml/Type.hs

```diff
@@ -35,26 +43,29 @@ newtype TableId = TableId
 -- TODO: describe how some TOML document will look like with this type
 {- | Represents TOML configuration value. -}
 data TOML = TOML
-    { tomlPairs       :: HashMap Key     Value
+    { tomlPairs       :: HashMap Key     AnyValue
     , tomlTables      :: HashMap TableId TOML
     , tomlTableArrays :: HashMap TableId (NonEmpty TOML)
     -- TODO: I don't really like the above structure, probably something better should be used
     }
 
+-- Needed for GADT parameterization
+data TomlType = TBool | TInt | TFloat | TString | TDate | TArray
```

Comment:
I came up with `TomlType` but personally I don't like this name.. Let's use something like `TValue` or `VType` or `ValueType` instead.